sway-core/src/asm_generation/fuel/fuel_asm_builder.rs

```diff
@@ -1249,6 +1250,40 @@ impl<'ir, 'eng> FuelAsmBuilder<'ir, 'eng> {
         Ok(())
     }
 
+    fn compile_get_global(
```

Comment:
`compile_get_global` does not cover the full IR semantics of `GlobalVar`s that do not need to be constants. As commented in the IR code, if a `GlobalVar` is mutable we do not put it into the data section.

Reply:
Makes sense. What I'll do here to check if a `GlobalVar` is mutable, and if so, leave that as a `todo!()`. The remaining code will then be correct as it's only dealing with global consts.